superset-frontend/packages/superset-ui-chart-controls/test/utils/getColorFormatters.test.ts

```diff
@@ -50,6 +50,8 @@ describe('getOpacity', () => {
     expect(getOpacity(100, 100, 50)).toEqual(0.05);
     expect(getOpacity(100, 100, 100, 0, 0.8)).toEqual(0.8);
     expect(getOpacity(100, 100, 50, 0, 1)).toEqual(0);
+    expect(getOpacity(999, 100, 50, 0, 1)).toEqual(1);
+    expect(getOpacity(-999, 100, 50, 0, 1)).toEqual(0.05);
```

Comment:
`Math.abs` inside the function will make sure `result` is never negative. We can replace this test with one that passes a high `minOpacity` like `0.9` and parameters that make `result` be less than this value to test `result < minOpacity`
```suggestion
```